### DISCUSSION TITLE
Multiple apps with Docker

### DISCUSSION THREAD

**Discussion by dv336699** [soketi/pws#270] [Q&A] [answered]
I can see that you can create multiple apps using a database like MySQL described in https://docs.soketi.app/app-management/sql-drivers/mysql which is great.

However, is there a way to pass multiple app id and secret using environment variables?

This is my docker-compose.yml file.

```
version: '3'
services:
  soketi:
    image: 'quay.io/soketi/soketi:0.17-16-alpine'
    environment:
      DEBUG: '1'
      METRICS_ENABLED: false
      METRICS_SERVER_PORT: '9601'
      DEFAULT_APP_ID: 'app-id'
      DEFAULT_APP_KEY: 'app-key'
      DEFAULT_APP_SECRET: 'app-secret'
      QUEUE_DRIVER: 'sync'
      DEFAULT_APP_ENABLE_CLIENT_MESSAGES: true
      DEFAULT_APP_MAX_CLIENT_EVENTS_PER_SEC: 10
    ports:
      - '127.0.0.1:6001:6001'
      - '127.0.0.1:9601:9601'
```

**Comment by rennokki** [comment, marked as answer]
With the environment variables, it's not possible. But you can [define a configuration file](https://docs.soketi.app/getting-started/environment-variables#file-configuration):

```json
{
    "appManager.array.apps": [
        { "id": ... },
        { "id: ... }
    ]
}
```

```bash
soketi start --config=/path/to/config.json
```

The app object looks like this:

```ts
export interface AppInterface {
    id: string;
    key: string;
    secret: string;
    maxConnections: string|number;
    enableClientMessages: boolean;
    enabled: boolean;
    maxBackendEventsPerSecond?: string|number;
    maxClientEventsPerSecond: string|number;
    maxReadRequestsPerSecond?: string|number;
    webhooks?: WebhookInterface[];
}
```

**Reply by Luigidefra** [reply]
Hi, I have some doubts about this.
If the data was added to the DB during its creation, why must it be read from the configuration file? Isn't it possible to read them directly from the DB at startup?

Another question
Is there an observer that listens if app rows are added or deleted from the DB?
